src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,162 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+from azure.cli.core.commands.parameters import name_type, tags_type, get_three_state_flag, get_enum_type, \
+    resource_group_name_type
+
+
+# pylint: disable=too-many-statements
+def load_arguments(self, _):
+    # synapse spark
+
+    for scope in ['batch', 'session', 'session-statement']:
+        with self.argument_context('synapse spark ' + scope) as c:
+            c.argument('workspace_name', help='The name of the workspace.')
+            c.argument('spark_pool_name', help='The name of the spark pool.')
+
+    for scope in ['synapse spark batch', 'synapse spark session']:
+        with self.argument_context(scope + ' create') as c:
+            c.argument('artifact_id', help='The artifact id.')
+            c.argument('job_name', arg_type=name_type, help='The spark batch or session job name.')
+            c.argument('file', help='The URI of file.')
+            c.argument('class_name', help='The class name.')
+            c.argument('args', nargs='+', help='The arguments of the job.')
+            c.argument('jars', nargs='+', help='The array of jar files.')
+            c.argument('files', nargs='+', help='The array of files URI.')
+            c.argument('archives', nargs='+', help='The array of archives.')
+            c.argument('conf', help='The configuration of spark batch job.')
+            c.argument('driver_memory', help='The memory of driver.')
+            c.argument('driver_cores', help='The number of cores in driver.')
+            c.argument('executor_memory', help='The memory of executor.')
+            c.argument('executor_cores', help='The number of cores in each executor.')
+            c.argument('num_executors', help='The number of executors.')
+            c.argument('tags', arg_type=tags_type, help='The tags of spark batch job.')
+            c.argument('detailed', arg_type=get_three_state_flag(),

```

Comment:
It seems this one should use `action='store_true'`. This is not a property of the resource, but a switch to control the command behavior, i.e. returning detailed response. More explains on the last part of [flags](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-flags).

Reply:
> It seems this one should use `action='store_true'`. This is not a property of the resource, but a switch to control the command behavior, i.e. returning detailed response. More explains on the last part of [flags](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-flags).

fixed